index.js

```diff
@@ -0,0 +1,106 @@
+const assert = require('assert');
+const debug = require('debug')('rubric');
+
+const { Rule } = require('./lib');
+
+class Rubric {
+  constructor() {
+    this.rules = [];
+    this.finalMap = null;
+  }
+
+  doesntMatter(_rule) {
+    const rule = Rule.from(_rule);
+    this.rules.push({ rule, type: 0 });
+    return this;
+  }
+
+  matters(_rule) {
+    const rule = Rule.from(_rule);
+    this.rules.push({ rule, type: 1 });
+    return this;
+  }
+
+  match(s1, s2) {
+    let s1prime = s1,
+      s2prime = s2;
+
+    // Collapse rules into finalMap and a set of functions.
+    // TODO: track the rules so that matching can be attributed back to rules
+    // that had an effect.
+    this.finalMap = new Map();
+    this.ruleFns = new Set();
+
+    this.rules.forEach(({ rule, type }) => {
+      assert(rule instanceof Rule);
+      // FIXME: do something more elegant than type 0 means matters, type 1
+      // means doesn't matter.
+      assert(typeof type === 'number');
+      assert(0 <= type <= 1);
```

Comment:
I believe we need `0 <= type && type <= 1`. `0 <= 999 <= 1` is `true`, for example.